activerecord/lib/active_record/associations/join_dependency.rb

```diff
@@ -92,7 +92,8 @@ def self.walk_tree(associations, hash)
       #    associations # => [:appointments]
       #    joins # =>  []
       #
-      def initialize(base, associations, joins)
+      def initialize(base, associations, joins, finder_methods_hash = {})
```

Comment:
Okay, the way I added the extra argument shouldn't break backwards compatibility due to the default argument but I understand if we'd want to avoid this if the interface is public.